test/agent/cmd/networking/tester/network.go

```diff
@@ -0,0 +1,267 @@
+		toContainerRoutes, err := netlink.RouteListFiltered(netlink.FAMILY_V4,
+			&netlink.Route{
+				Dst: mainTableRules[0].Dst,
+			}, netlink.RT_FILTER_DST)
+		if err != nil {
+			fmt.Errorf("failed to find ip rule with destination %s: %v", podIP.String(), err)
+		}
+
+		if len(toContainerRoutes) != 1 {
+			validationErrors = append(validationErrors,
+				fmt.Errorf("found 0 or more than 1 route to the container %s: %v",
+					podIP.String(), toContainerRoutes))
+			continue
+		}
+
+		// Verify that the link index for the route is the same as the veth pair index
+		if toContainerRoutes[0].LinkIndex != link.Attrs().Index {
+			validationErrors = append(validationErrors,
+				fmt.Errorf("the link index for to contianer route %d is different from"+
+					" veth index %d", toContainerRoutes[0].LinkIndex, link.Attrs().Index))
+			continue
+		}
+
+		log.Printf("found route to %v with link index %d",
+			toContainerRoutes[0].Dst, toContainerRoutes[0].LinkIndex)
+
+		// Pod with IP from Secondary ENI will have additional rule for destination to each
+		// VPC Cidr block
+		if pod.IsIPFromSecondaryENI {
+			var destinationCidrRange []string
+			for _, rule := range nonMainTableRules {
+				destinationCidrRange = append(destinationCidrRange, rule.Dst.String())
+				// Add the secondary route table Index, we will do validation later
+				secondaryRouteTableIndex[rule.Table] = true
+			}
+			sort.Strings(destinationCidrRange)
+			// Validate there is one rule for each VPC CIDR Range
+			if !reflect.DeepEqual(destinationCidrRange, podNetworkingValidationInput.VPCCidrRange) {
+				validationErrors = append(validationErrors,
+					fmt.Errorf("failed to find route to vpc cidr: %v", destinationCidrRange))
+				continue
+			}
+			log.Printf("validated all routes for VPC CIDR %v are present for secondary ENI",
+				destinationCidrRange)
+		}
+		log.Printf("validation for pod %s/%s succeeded", pod.PodNamespace, pod.PodName)
+	}
+
+	// Finally validate that the route table for secondary ENI has the right routes
+	for index, _ := range secondaryRouteTableIndex {
+		routes, err := netlink.RouteListFiltered(netlink.FAMILY_V4,
+			&netlink.Route{
+				Table: index,
+			}, netlink.RT_FILTER_TABLE)
+		if err != nil {
+			validationErrors = append(validationErrors,
+				fmt.Errorf("failed to find route for table with index %d:%v", index, err))
+			continue
+		}
+
+		// Route 1 should route all traffic through Gateway via Secondary ENI
+		gateway := routes[0].Gw
+		secondaryENIIndex := routes[0].LinkIndex
+
+		// Route 2 should route all traffic intended for Gateway IP through Secondary ENI
+		if !routes[1].Dst.IP.Equal(gateway) ||
+			routes[1].LinkIndex != secondaryENIIndex {
+			validationErrors = append(validationErrors,
+				fmt.Errorf("found invalid route for secondary ENI %v", err))
+			continue
+		}
+		log.Printf("validated route table for secondary ENI %d has right routes", index)
+	}
+
+	return validationErrors
```

Comment:
Would be nice to have a struct where you can define pod and failure type(s) - will make debugging easier